pkg/router/template/router.go

```diff
@@ -490,51 +521,38 @@ func (r *templateRouter) AddRoute(id string, route *routeapi.Route, host string)
 	r.lock.Lock()
 	defer r.lock.Unlock()
 
-	frontend, _ := r.findMatchingServiceUnit(id)
+	frontend, _ := r.findMatchingServiceUnit(serviceID)
 
 	//create or replace
-	frontend.ServiceAliasConfigs[backendKey] = config
-	r.state[id] = frontend
-	r.cleanUpdates(id, backendKey)
+	config.ServiceUnitNames[frontend.Name] = weight
+	r.state[backendKey] = config
+	r.serviceUnits[serviceID] = frontend
+	//r.cleanUpdates(serviceID, backendKey)
 	return true
 }
 
-// cleanUpdates ensures the route is only under a single service key.  Backends are keyed
```

Comment:
No need for the cleanup now.. we have got two tables where the object is referenced. So we do not need to delete the duplicate objects. Takes care of the situation where a route is created before a service is created too.